mlflow/utils/warnings_utils.py

```diff
@@ -23,3 +29,27 @@ def color_warning(message: str, stacklevel: int, color: str, category: type[Warn
         category=category,
         stacklevel=stacklevel + 1,
     )
+
+
+def suppress_warnings_containing(pattern):
```

Comment:
I'd just wrap `raw_response = ws_client.api_client.do`

```
            with warnings.catch_warnings():
                warnings.filterwarnings("ignore", message=f"retry_after_secs")
                raw_response = ws_client.api_client.do(...)
```

Reply:
I'd consider adding a helper function if we need to do the same > 5 places.

Reply:
How many lines can this save? Just one

```
with warnings.catch_warnings():
     warnings.filterwarnings("ignore", message=f"retry_after_secs")
     ...
```

vs.

```
with suppress_warnings_containing(...):
     ...
```

Reply:
Personally I like decorator more than context manager, so we don't need to indent everything 😅 
What's the concern of adding this helper function though? Maintenance burden?

Reply:
After offline discussion: the context manager helps us locate which line emits the warning. Updated